File: TransCoder_tokenized_test_set_functions/java/MAXIMUM_DISTANCE_TWO_OCCURRENCES_ELEMENT_ARRAY.java
static int maxDistance ( int [ ] arr, int n ) {
  HashMap < Integer, Integer > map = new HashMap < > ( );
  int max_dist = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( ! map . containsKey ( arr [ i ] ) ) map . put ( arr [ i ], i );
    else max_dist = Math . max ( max_dist, i - map . get ( arr [ i ] ) );
  }
  return max_dist;
}

File: results/java/MAXIMUM_DISTANCE_TWO_OCCURRENCES_ELEMENT_ARRAY.cpp
int f_filled ( int * arr , int n ) {
  unordered_map < int , int > map ;
  int maxDist = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( ! map . count ( arr [ i ] ) ) {
      map [ arr [ i ] ] = i ;
    }
    else {
      maxDist = max ( maxDist , i - map [ arr [ i ] ] ) ;
    }
  }
  return maxDist ;
}
